src/auth/impersonated.ts

```
@@ -253,4 +268,27 @@ export class Impersonated extends OAuth2Client implements IdTokenProvider {
 
     return res.data.token;
   }
+
+  /**
+   * Fetches a trustBoundary for the given service account.
+   * @param authHeader the authheader for calling the lookup endpoint
+   */
+  async fetchTrustBoundary(
+    authHeader: string,
+  ): Promise<TrustBoundaryData | null> {
+    if (!this.targetPrincipal) {
```

We should throw an error if target principal is missing.